test/e2e/autoscale_test.go

```diff
@@ -289,3 +289,7 @@ func TestAutoscaleUpDownUp(t *testing.T) {
 			   up. %s`, deploymentName, err)
 	}
 }
+
+func TestAutoscaleExceedsLimits(t *testing.T) {
```

Comment:
I don't think we need to test in the autoscaler; the resource limits supplied are per-pod, so autoscaling will just make more pods.

At some point, we'll need to put in a circuit-breaker that prevents scaling up if existing pods are crash-looping. @josephburnett to track.

Reply:
To be honest, my intention was to test the ScaleUpAndDown behavior if some resources would be freed or allocated. If pods would be created or destroyed.  But that's probably not the responsibility of the autoscaler itself.